sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSampledSpansProcessor.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace.export;
+
+import com.google.common.util.concurrent.MoreExecutors;
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * Implementation of the {@link SpanProcessor} that batches spans exported by the SDK then pushes
+ * them to the exporter pipeline.
+ *
+ * <p>All spans reported by the SDK implementation are first added to a synchronized queue (with a
+ * {@code maxQueueSize} maximum size, after the size is reached spans are dropped) and exported
+ * every {@code scheduleDelayMillis} to the exporter pipeline in batches of {@code
+ * maxExportBatchSize}.
+ *
+ * <p>If the queue gets half full a preemptive notification is sent to the worker thread that
+ * exports the spans to wake up and start a new export cycle.
+ *
+ * <p>This batch {@link SpanProcessor} can cause high contention in a very high traffic service.
+ * TODO: Add a link to the SpanProcessor that uses Disruptor as alternative with low contention.
+ */
+public final class BatchSampledSpansProcessor implements SpanProcessor {
+  private static final String WORKER_THREAD_NAME = "BatchSampledSpansProcessor_WorkerThread";
+  private final Worker worker;
+  private final Thread workerThread;
+
+  private BatchSampledSpansProcessor(
+      SpanExporter spanExporter,
+      long scheduleDelayMillis,
+      int maxQueueSize,
+      int maxExportBatchSize) {
+    this.worker = new Worker(spanExporter, scheduleDelayMillis, maxQueueSize, maxExportBatchSize);
+    this.workerThread = newThread(worker);
+    this.workerThread.start();
+  }
+
+  @Override
+  public void onStartSync(ReadableSpan span) {}
```

Comment:
Nit: add a TODO here to track the started spans?

Reply:
We will implement a different SpanProcessor for active spans. I think with this design we can have independent modules for these separate functionalities.